ksa_compliance/ksa_compliance/test/test_zatca_state3_integration.py

```
@@ -1517,6 +1517,90 @@ def test_rounding_with_multiple_line_items(self):
         frappe.logger().info("   ✓ Multi-line item rounding is correct")
         frappe.logger().info("✅ test_rounding_with_multiple_line_items completed")
 
+    def test_tax_calculation_row_wise_precision(self):
+        """
+        Tests row-wise tax rounding with the specific amounts that triggered the
+        advance-payment over-allocation bug (total_allocated > grand_total by 0.01).
+
+        Rates: 893.25, 28.5, 250 — VAT 15%
+        Per-item rounding:
+          893.25 × 0.15 = 133.9875 → 133.99
```

Ruff (0.15.14)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)